### PR TITLE
Enable FxCop - part 03

### PR DESCRIPTION
**Pull request by eddynaka**
Refs #992

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Api/Context/ThreadLocalRuntimeContextSlot.cs [hunk 1]
```diff
@@ -23,9 +24,10 @@ namespace OpenTelemetry.Context
     /// The thread local (TLS) implementation of context slot.
     /// </summary>
     /// <typeparam name="T">The type of the underlying value.</typeparam>
-    public class ThreadLocalRuntimeContextSlot<T> : RuntimeContextSlot<T>
+    public class ThreadLocalRuntimeContextSlot<T> : RuntimeContextSlot<T>, IDisposable
```
Comment (reyang) [inline src/OpenTelemetry.Api/Context/ThreadLocalRuntimeContextSlot.cs:27]: In this case we probably can put the `IDisposable` in the base class.

Comment (eddynaka) [reply src/OpenTelemetry.Api/Context/ThreadLocalRuntimeContextSlot.cs:27]: thanks!

File: src/OpenTelemetry.Api/Trace/SpanHelper.cs [hunk 2]
```diff
@@ -141,7 +139,7 @@ public static Status ResolveCanonicalCodeToStatus(string statusCanonicalCode)
 
             if (!success)
             {
-                return DefaultStatus;
+                return default;
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Trace/SpanHelper.cs:142]: @rajkumar-rangaraj mentioned once that returning default is not as faster as returning the static DefaultStatus.. not sure if its relevant, as Status is struct...

Comment (eddynaka) [reply src/OpenTelemetry.Api/Trace/SpanHelper.cs:142]: i will try some things and pst here the benchmark! :)

Comment (eddynaka) [reply src/OpenTelemetry.Api/Trace/SpanHelper.cs]: i rollback my changes. the default variable is a little better.

File: src/OpenTelemetry.Api/Context/CorrelationContext.cs [hunk 3]
```diff
@@ -72,5 +86,17 @@ public bool Equals(CorrelationContext other)
...
+        /// <inheritdoc/>
+        public override bool Equals(object obj)
+        {
+            return obj is CorrelationContext context && this.Equals(context);
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            throw new NotImplementedException();
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Context/CorrelationContext.cs:99]: `this.entries.GetHashCode` is probably better than a throw here.

File: src/OpenTelemetry.Api/Context/DistributedContext.cs [hunk 4]
```diff
@@ -81,5 +101,17 @@ public bool Equals(DistributedContext other)
...
+        /// <inheritdoc/>
+        public override bool Equals(object obj)
+        {
+            return obj is DistributedContext context && this.Equals(context);
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            throw new NotImplementedException();
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Context/DistributedContext.cs:114]: Maybe `this.correlationContext.GetHashCode()` will work here?

**Comment by eddynaka** [comment]
@CodeBlanch , just updated both. thanks!

File: src/OpenTelemetry.Api/Context/RuntimeContext.cs [hunk 5]
```diff
@@ -23,7 +23,7 @@ namespace OpenTelemetry.Context
     /// <summary>
     /// Generic runtime context management API.
     /// </summary>
-    public sealed class RuntimeContext
+    public static class RuntimeContext
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Context/RuntimeContext.cs:26]: why this change?

Comment (eddynaka) [reply src/OpenTelemetry.Api/Context/RuntimeContext.cs:26]: Since it just have static methods, it was expecting to be static class as well. I can revert if needed.

Comment (reyang) [reply src/OpenTelemetry.Api/Context/RuntimeContext.cs:26]: Seems static fits better in this case.
